CodeLearn/Component/DiscussionHome.razor

```diff
@@ -0,0 +1,129 @@
+    <MudContainer Style="padding-top: 5%;">
+    @{
+        for (int count = (_pageSelected * _sizePage) - _sizePage; count < (_pageSelected * _sizePage) - 1; count++)
+            if (count < _searchedDiscussion.Count())
+            {
+                int i = count;
+            <MudCard>
+                <MudCardHeader Style="padding-bottom: 0px;">
+                    <CardHeaderContent>
+                        <MudAvatar Color="Color.Info" Variant="Variant.Outlined">@GetUserNamebyID(_searchedDiscussion[i].UserId.Value)[0]</MudAvatar>
+                        <MudLink Typo="Typo.h5" Href="@("/DetailDiscussion/" + _searchedDiscussion[i].Id)" Color="Color.Dark">@_searchedDiscussion[i].Question</MudLink>
+                    </CardHeaderContent>
+                </MudCardHeader>
+                <MudCardContent Style="padding-top: 0px; margin-left: 45px;">
+                    <MudText Style="max-width: 67%;display: -webkit-box;-webkit-box-orient: vertical; -webkit-line-clamp: 3;overflow:hidden; text-overflow: ellipsis;" Typo="Typo.body2">@_searchedDiscussion[i].Content</MudText>
+                </MudCardContent>
+                <MudText Style="@($"color:{Colors.BlueGrey.Lighten4}; font-size: 13px;  margin-left: 45px;")">
+                    &emsp;
+                    Người đăng: <MudElement HtmlTag="a"
+                                            Style="@($"color:{Colors.BlueGrey.Lighten2};  font-weight:bold;")">
+                        @GetUserNamebyID(_searchedDiscussion[i].UserId.Value)
+                    </MudElement>
+                    &emsp;Ngày đăng: <MudElement HtmlTag="a"
+                                                 Style="@($"color:{Colors.BlueGrey.Lighten2};  font-weight:bold;")">
+                        @_searchedDiscussion[i].CreateOn
+                    </MudElement>
+                </MudText>
+                <MudChipSet Style="margin-left: 53px;">
+                    @foreach (var value in _searchedDiscussion[i].HashTag)
+                    {
+                    <MudChip Style="@($"background-color :{Colors.BlueGrey.Lighten4};")" Text="@value"></MudChip>
+                    }
+                </MudChipSet>
+            </MudCard>
+            <MudDivider />
+            <MudDivider />
+            <br \>
+        }
+    }
+    </MudContainer>
+</MudPaper>
+<MudPagination Style="margin-top: 14px;" Color="Color.Primary" Count="(_searchedDiscussion.Count() / _sizePage) + 1" @bind-Selected="@_pageSelected" />
+@code 
+{
+    [CascadingParameter]
+    private Task<AuthenticationState> _authenticationStateTask { get; set; }
+    private int _pageSelected = 1, _sizePage = 7;
+    private string _searchString = String.Empty;
+    CodeLearn.Models.User _user = new();
+    IList<Discussion> _discussions, _searchedDiscussion;
+    bool _isLogIn;
+    protected override void OnInitialized()
+    {
+        _discussions = DiscussionRepository.GetAllDiscussionType().OrderByDescending(t => t.CreateOn).ToList();
```

Comment:
Không nên get hết toàn bộ data về rồi mới xử lý paging. Trường hợp có nhiều record sẽ phải xử lý rất nặng.
Dùng Skip và Take để xử lý paging trên db luôn.